libs/ts-rest/core/src/lib/client.ts

```diff
@@ -352,7 +350,8 @@ export const getRouteQuery = <TAppRoute extends AppRoute>(
       },
     });
 
-    if (!clientArgs.throwOnUnknownStatus) {
+    const strictMode = route.strict ?? clientArgs.throwOnUnknownStatus;
```

Comment:
Perhaps `throwOnUnknownStatus` can be removed in a follow up PR?

Reply:
Since the default behavior is not throw on any response, it might catch some people off guard if we start throwing when strict mode is enabled. I think we should only throw on `throwOnUnknownStatus` and not strict mode. 

I know it's technically a non-breaking change, but just changing an option in the contract is not immediately obvious that it might affect the runtime behavior of the client, so we should keep strict mode for type-safety only for now.

Reply:
Perhaps some type docs would help in this case? That way they get information in their text editor if it is setup to see the method docs. 

Something that your comment made me wonder was what we should do on the server side if there is an invalid response. Should it also support this?

If you want to go down this path then perhaps a strict mode option would be better? For example `strict: throwOnUnknownStatus` and this would apply on both sides?

Reply:
Well yeah the main point from this strict mode is to narrow down the types to exclude all undefined status codes and the corresponding `unknown` body type from the responses. So you should get a type error if you try to return a response from the server with a status that is not defined in the contract. So the complement to that is to only narrow the type on the client side as well and do nothing else.

ts-rest's philosophy is to be as light and unobtrusive as possible, so we should target to enforce this only through type-safety by default. Any extra "guardrails" that a developer might want should be optional and explicitly enabled rather than enforcing it by default or even not having the option to turn it off.

Reply:
This makes sense. Thanks for that context.

With this in mind I suggest two strict modes:
- `strict: 'responseTypes'`
- `strict: 'throwOnUnknownResponse'`

This allows the cognitive load for users to be kept to a minimum, due to consolidated configuration, while allowing them to choose the behaviour that they want. This would replace `throwOnUnknownStatus`. If in the future this causes any unforeseen issues in the future then there is a way to add additional modes. What do you think? If you don't like it then happy to go with `strict: true` and `throwOnUnknownStatus`.

Reply:
Yeah that sounds very reasonable. I think there's also the point that the current change would only affect the vanilla client, but not the other clients tanstack query clients since they already have their own error handling, so it wouldn't really make sense to have an option in the contract itself that's only specific to one client, so to me it makes sense that the "throw" option should live in that client's options rather than in the contract.

Reply:
Changed it to only throw if  `throwOnUnknownStatus` is set.